overlord/snapstate/export_test.go

```diff
@@ -471,3 +471,47 @@ func MockCgroupMonitorSnapEnded(f func(string, chan<- string) error) func() {
 		cgroupMonitorSnapEnded = old
 	}
 }
+
+func CalculateTaskProgress(st *state.State, snapName string, instanceName string, revision snap.Revision) (int, int, string) {
+	return calculateTaskProgress(st, snapName, instanceName, revision)
+}
+
+func NotifyBeginRefresh(st *state.State, taskCount int, instanceName string, revision snap.Revision, snapName string, desktopPath string) {
+	notifyBeginRefresh(st, taskCount, instanceName, revision, snapName, desktopPath)
+}
+
+func SnapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
+	return snapSetupAndState(t)
+}
+
+func MockAsyncBeginDeferredRefreshNotification(f func(context context.Context, client *userclient.Client, finishInfo *userclient.FinishedSnapRefreshInfo)) func() {
+	old := asyncBeginDeferredRefreshNotification
+	asyncBeginDeferredRefreshNotification = f
+	return func() {
+		asyncBeginDeferredRefreshNotification = old
+	}
```

Comment:
For the remaining the pattern can be this
```suggestion
	r := testutil.Backup(&asyncBeginDeferredRefreshNotification)
	asyncBeginDeferredRefreshNotification = f
	return r
```

Reply:
But all the other mock functions do use the old method... Do you want another MR that updates to this new method?